### DISCUSSION TITLE
Promise rejected for "unexpected end of file" for some products

### DISCUSSION THREAD

**Discussion by Peariforme** [Ephigenia/ikea-availability-checker#97] [Q&A] [answered]
Hello,

When I use the checker.availability function with buCode = 199 and ProductID = 69444085 I get the following error thrown by the httpAdapter function :

Une exception s'est produite : Error: unexpected end of file
  at Zlib.zlibOnError [as onerror] (node:zlib:190:17)
    at Zlib.callbackTrampoline (node:internal/async_hooks:130:17)

I am not getting this error for every product. For example in this list
394.440.20
292.897.84
092.897.61
294.141.27
504.688.54
393.255.31
694.440.85

Only 504.688.54 is working fine. Is there something in the Ikea answer which is not correctly parsed ? I didn't found a way to follow the Ikea answer as I am not used to async function.

I tested directly threw the console command npx ikea-availability-checker -- stock --country fr 69444085 and it seems to work fine.
I don't know what do to now to solve my problem. Someone did get this problem already ?

Thank you.
Sincerely,
Peariforme

**Comment by Ephigenia** [comment, marked as answer]
I was checking the product codes that you mentioned. They all (except 504.688.54) require the "S" in front of it. Unfortunately some of the IKEA products have an "S" in front. That is only visible in the URL of the product(s). Taken the first product from the list "394.440.20" and searching it on the ikea.fr website leads to this product: https://www.ikea.com/fr/fr/p/aepplaroe-bain-de-soleil-teinte-brun-froesoen-duvholmen-beige-s39444020/

You’ll get the stock numbers by running:
```
npx ikea-availability-checker stock --country fr s39444020    
```
*(If you’re running the command using `npx` you can leave out the `--`, they are only required when you run the script via `npm`)*

**Reply by Peariforme** [reply]
Thank you for your help, this is indeed solving the problem.
